integration-tests/api/__tests__/store/products.js

```diff
@@ -1006,5 +1006,35 @@ describe("/store/products", () => {
         "type",
       ])
     })
+
+    it("response contains only fields defined with `fields` param including the relation", async () => {
+      const api = useApi()
+
+      const response = await api.get(
+        "/store/products/test-product?fields=id,variants.title"
+      )
+
+      expect(response.status).toEqual(200)
+
+      expect(Object.keys(response.data.product)).toEqual([
```

Comment:
suggestion: To ensure that this isn't slipped in the future, we could create a test util that accepts allowed**fields param and tests each of them in the response?

Reply:
Yes indeed i ll add that 👌 still fixing some stuff at the moment 😇